internal/server/v1/dlq/handler_test.go

```diff
@@ -0,0 +1,231 @@
+	panic("")
+}
+
+func (*mockEntropyService) GetLog(context.Context, *entropyv1beta1.GetLogRequest, ...grpc.CallOption) (entropyv1beta1rpc.ResourceService_GetLogClient, error) {
+	panic("")
+}
+
+func (*mockEntropyService) GetResourceRevisions(context.Context, *entropyv1beta1.GetResourceRevisionsRequest, ...grpc.CallOption) (*entropyv1beta1.GetResourceRevisionsResponse, error) {
+	panic("")
+}
+
+type mockGcsClient struct {
+	mock.Mock
+}
+
+func (g *mockGcsClient) ListTopicDates(bucketInfo gcs.BucketInfo) (map[string]map[string]int64, error) {
+	args := g.Called(bucketInfo)
+	if args.Get(0) == nil {
+		err, _ := args.Get(1).(error)
+		return nil, err
+	}
+	topicDates, _ := args.Get(0).(map[string]map[string]int64)
+	return topicDates, nil
+}
+
+type mockHTTPWriter struct {
+	messages []string
+}
+
+func (*mockHTTPWriter) Header() http.Header {
+	return http.Header{}
+}
+
+func (m *mockHTTPWriter) Write(bytes []byte) (int, error) {
+	m.messages = append(m.messages, string(bytes[:]))
+	return 0, nil
+}
+
+func (*mockHTTPWriter) WriteHeader(int) {
+}
+
+func TestListTopicDates(t *testing.T) {
+	eService := &mockEntropyService{}
+	gClient := &mockGcsClient{}
+	handler := NewHandler(NewService(eService, gClient))
+	httpWriter := &mockHTTPWriter{}
+	httpRequest := &http.Request{}
+	config := &entropy.Config{
+		Stopped:      false,
+		StopTime:     nil,
+		Replicas:     0,
+		Namespace:    "",
+		DeploymentID: "",
+		EnvVariables: map[string]string{
+			firehose.ConfigDLQBucket:          "test-bucket",
+			firehose.ConfigDLQDirectoryPrefix: "test-prefix",
+		},
+		ResetOffset:   "",
+		Limits:        entropy.UsageSpec{},
+		Requests:      entropy.UsageSpec{},
+		Telegraf:      nil,
+		ChartValues:   nil,
+		InitContainer: entropy.InitContainer{},
+	}
+	configProto, _ := utils.GoValToProtoStruct(config)
+	eService.On(
+		"GetResource",
+		context.Background(),
+		mock.Anything,
+		[]grpc.CallOption(nil)).Return(
+		&entropyv1beta1.GetResourceResponse{
+			Resource: &entropyv1beta1.Resource{
+				Urn:       "",
+				Kind:      "",
+				Name:      "",
+				Project:   "",
+				Labels:    nil,
+				CreatedAt: nil,
+				UpdatedAt: nil,
+				Spec: &entropyv1beta1.ResourceSpec{
+					Configs:      configProto,
+					Dependencies: nil,
+				},
+				State:     nil,
+				CreatedBy: "",
+				UpdatedBy: "",
+			},
+		}, nil)
+	topicDates := make(map[string]map[string]int64)
+	topicDates["topic-1"] = make(map[string]int64)
+	topicDates["topic-2"] = make(map[string]int64)
+	topicDates["topic-1"]["2023-08-26"] = int64(1234)
+	topicDates["topic-1"]["2023-12-10"] = int64(4321)
+	topicDates["topic-2"]["2023-09-20"] = int64(99)
+	gClient.On("ListTopicDates", gcs.BucketInfo{
+		BucketName: "test-bucket",
+		Prefix:     "test-prefix",
+		Delim:      "",
+	}).Return(topicDates, nil)
+	handler.listFirehoseDLQ(httpWriter, httpRequest)
+	expectedMap := make(map[string]map[string]map[string]int64)
+	_ = json.Unmarshal([]byte(httpWriter.messages[0]), &expectedMap)
```

Comment:
let's always try to handle error. in this case, we can easily do
```suggestion
	err = json.Unmarshal([]byte(httpWriter.messages[0]), &expectedMap)
	require.NoError(t, err)
```

this one line can save us a few mins debugging